jormungandr/src/network/p2p/topology.rs

```diff
@@ -166,6 +166,21 @@ impl P2pTopology {
         })
     }
 
+    pub fn list_available_limit<E>(
+        &self,
+        n: usize,
+    ) -> impl Future<Item = Vec<poldercast::Node>, Error = E> {
+        self.read().map(move |topology| {
+            topology
+                .nodes()
+                .all_available_nodes()
+                .into_iter()
+                .take(n)
```

Comment:
This will give everybody the same list of peers, no randomization?

Reply:
no idea but it's indeed quite possible indeed, I didn't really have time to think about how to randomize this access (at it rightly should)

Reply:
use `view` instead